src/CodeGen_Hexagon.cpp

```diff
@@ -702,9 +706,24 @@ void CodeGen_Hexagon::visit(const Mul *op) {
             std::swap(a, b);
 
         // Try to find an intrinsic for one of the operands being a scalar.
+        // All the non-widening vector by scalar multiplies expect a narrower
+        // scalar. Only two narrow types work, 8 bit and 16 bit. Start with
+        // the 8 bit ones
+        Expr narrow_b = lossless_cast(b.type().with_bits(8), b);
```

Comment:
How about:

    for (int bits : {8, 16}) {
        Expr narrow_b = lossless_cast(b.type().with_bits(bits), b);
        Expr op_b = narrow_b.defined() ? narrow_b : b;
        value = call_intrin(op->type,
                             "halide.hexagon.mpyi" +
                             type_suffix(a, opb),
                             {a, opb},
                              true /*maybe*/);
        if (value) return;
    }

?